src/layouts/EditPage/EditPage.tsx

```diff
@@ -0,0 +1,260 @@
+import PagePreview from "components/pages/PagePreview"
+
+import { EditorContextProvider } from "contexts/EditorContext"
+import { EditorModalContextProvider } from "contexts/EditorModalContext"
+import { ServicesContext } from "contexts/ServicesContext"
+
+import { useGetPageHook, useUpdatePageHook } from "hooks/pageHooks"
+import { useGetSiteColorsHook } from "hooks/settingsHooks"
+import useRedirectHook from "hooks/useRedirectHook"
+
+import Iframe from "layouts/components/Editor/Iframe"
+
+import { MediaService } from "services/MediaService"
+
+import { getImageDetails } from "utils/images"
+import { isWriteActionsDisabled } from "utils/reviewRequests"
+import { createPageStyleSheet } from "utils/siteColorUtils"
+
+// Isomer utils
+
+import { getDecodedParams } from "utils"
+
+import "easymde/dist/easymde.min.css"
+
+import { Editor } from "../components/Editor/Editor"
+
+import { DEFAULT_BODY } from "./constants"
+
+// axios settings
+axios.defaults.withCredentials = true
+
+interface EditPageLayoutProps {
+  getPageBody: () => string
+}
+
+const EditPageLayout = ({
+  getPageBody,
+  children,
+}: PropsWithChildren<EditPageLayoutProps>) => {
+  const params = useParams<{ siteName: string }>()
+  const decodedParams = getDecodedParams(params)
+  const {
+    mutateAsync: updatePageHandler,
+    isLoading: isSavingPage,
+  } = useUpdatePageHook(params, {
+    // NOTE: Not deleting this as this is important enough
+    // to leave here so that we avoid regression.
+    // onError: (err) => {
+    //   if (err.response.status === 409) onOverwriteOpen()
+    // },
+  })
+
+  const { siteName } = decodedParams
+
+  const { setRedirectToNotFound } = useRedirectHook()
+  const { data: pageData, isLoading: isLoadingPage } = useGetPageHook(params, {
+    onError: () => setRedirectToNotFound(siteName),
+  })
+  const { data: siteColorsData } = useGetSiteColorsHook(params)
+
+  const isWriteDisabled = isWriteActionsDisabled(siteName)
+
+  useEffect(() => {
+    if (siteColorsData)
+      createPageStyleSheet(
+        siteName,
+        siteColorsData.primaryColor,
+        siteColorsData.secondaryColor
+      )
+  }, [siteColorsData, siteName])
+
+  return (
+    <Greyscale isActive={isWriteDisabled}>
+      <Flex flexDir="column" h="full">
+        <Header
+          title={pageData?.content?.frontMatter?.title || ""}
+          // TODO: Add this check back in dynamically
+          shouldAllowEditPageBackNav
+          isEditPage
+          params={decodedParams}
+        />
+        <Flex flexDir="row" w="100%" h="100%">
+          {/* Editor */}
+          {children}
+        </Flex>
+        <Spacer />
+        <Footer>
+          <Button
+            onClick={() => {
+              updatePageHandler(({
+                pageData: {
+                  frontMatter: pageData?.content?.frontMatter,
+                  pageBody: getPageBody(),
+                  sha: pageData.sha,
+                },
+              } as unknown) as void)
```

Comment:
why do we need to type this to void ah?

Reply:
the underlying hook is in jsx, which has auto inferred types.